### PR TITLE
Moderation Session docs update

### PR DESCRIPTION
**Pull request by stevenGravy**
Provides:
 - fleshed out roles examples.  Previous examples would not work with being created
 - clarify on how to join
 - example of joining a moderated session.

### CODE REVIEW

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 1]
```diff
@@ -65,27 +65,40 @@ The following fields are optional for `require_session_join`:
...
-#### Example
+#### Example Session Moderation Required Role
 
 The policy below specifies that the `prod-access` role must have a minimum of
 two users with the role `auditor` and the mode `moderator` present in the
 session to start it. The policy applies to SSH and Kubernetes sessions only.
 
 When a user with this require policy starts a session, it will be pending until
-the policy is fulfilled. 
+the policy is fulfilled. The role matches to `env: prod` labeled `ssh` and `k8s`
```
Comment (zmb3) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:75]: nit: `ssh` and `k8s` aren't resources, they're session kinds.

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 2]
```diff
@@ -65,27 +65,40 @@ The following fields are optional for `require_session_join`:
 |---|---|---|
 |`on_leave`|[On leave](#on-leave)|The action to take when the policy is no longer satisfied|
 
-#### Example
+#### Example Session Moderation Required Role
```
Comment (zmb3) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:68]: This reads like a proper noun, but we don't really have a first class resource called "Session Moderation Required Role".

I would personally just keep `Example` here.

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 3]
```diff
@@ -65,27 +65,40 @@ The following fields are optional for `require_session_join`:
...
 When a user with this require policy starts a session, it will be pending until
-the policy is fulfilled. 
+the policy is fulfilled. The role matches to `env: prod` labeled `ssh` and `k8s`
+resources. 
 
 ```yaml
 kind: role
 metadata:
   name: prod-access
+version: v5
```
Comment (zmb3) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:82]: 🤦 great catch. this is critical!

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 4]
```diff
@@ -65,27 +65,40 @@ The following fields are optional for `require_session_join`:
...
+#### Example Session Moderation Required Role
 
 The policy below specifies that the `prod-access` role must have a minimum of
 two users with the role `auditor` and the mode `moderator` present in the
 session to start it. The policy applies to SSH and Kubernetes sessions only.
 
 When a user with this require policy starts a session, it will be pending until
-the policy is fulfilled. 
+the policy is fulfilled. The role matches to `env: prod` labeled `ssh` and `k8s`
+resources.
```
Comment (zmb3) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:77]: Let's add a plain-text description of what this policy is doing. Something like:

> This policy requires that at least one user with the `auditor` role is present in order for SSH or Kubernetes sessions to begin.

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 5]
```diff
@@ -105,7 +118,13 @@ The following are required options for `join_sessions`:
 |`kinds`|`[]`[Session kind](#session-kinds)|The kind of session that the policy applies to|
 |`modes`|`[]`[Participant mode](#participant-modes)|The participant mode that applies to the user joining the Moderated Session under this policy|
 
-#### Example
+<Admonition type="note">
+Moderator or observers can only join via CLI currently with `tsh join --mode=observer|moderator <session id>`.  Web joining is
```
Comment (zmb3) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:122]: I would rephrase this a bit.

> In order to join a session in `moderator` or `observer` mode, users must join
> using the CLI: `tsh join --mode=observer|moderator`.
> Joining from the web UI always joins in `participant` mode.

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 6]
```diff
@@ -124,6 +144,39 @@ spec:
         modes: ['moderator', 'observer']
 ```
 
+#### Joining sessions example
+
+Here is an example of Jeff with role `prod-access` joining to a 
+production environment.
```
Comment (zmb3) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:150]:
```suggestion
Here is an example of Jeff with role `prod-access` connecting to
a SSH server in the production environment.
```

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 7]
```diff
@@ -124,6 +144,39 @@ spec:
...
+Teleport > Creating session with ID: 46e2af03-62d6-4e07-a886-43fe741ca044...
+Teleport > Controls
+  - CTRL-C: Leave the session
+  - t: Forcefully terminate the session (moderators only)
+Teleport > User jeff joined the session.
+Teleport > Waiting for required participants...
+```
+
+Now Alice with the `auditor` role joins as a moderator and
+the session can begin.
```
Comment (zmb3) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:163]:
```suggestion
Jeff's session is paused, waiting for the required observers.

Now Alice with the `auditor` role joins as a moderator and
the session can begin. 
```

**Review by alexfornuto — APPROVED** [review]
EDIT: Disregard, was looking at a single commit.

File: docs/pages/access-controls/guides/moderated-sessions.mdx (hunk 2 above)
Comment (alexfornuto) [reply docs/pages/access-controls/guides/moderated-sessions.mdx:68]: Or the confusion here may be because we don't use title case in our titles, for whatever reason.

**Comment by alexfornuto** [comment]
As part of the docs team's effort to sort and prioritize docs updates I've added this to our queue and assigned @stevenGravy, since it looks like there are outstanding review comments to be addressed by the author.

Alternately Steven, let us know if you would like the docs team to pick up from here and address the remaining notes.

File: docs/pages/access-controls/guides/moderated-sessions.mdx (hunk 2 above)
Comment (ptgott) [reply docs/pages/access-controls/guides/moderated-sessions.mdx:68]: This is to make it clearer which terms in a heading are proper nouns

**Comment by ptgott** [comment]
@alexfornuto @stevenGravy Just following up on the status of this one. Let me know if you need another review!

**Comment by stevenGravy** [comment]
> @alexfornuto @stevenGravy Just following up on the status of this one. Let me know if you need another review!

Please take a look when you can @ptgott .

**Comment by alexfornuto** [comment]
@stevenGravy it looks to me like this still has outstanding review items to be addressed by the author.

File: docs/pages/access-controls/guides/moderated-sessions.mdx (hunk 5 above)
Comment (stevenGravy) [reply docs/pages/access-controls/guides/moderated-sessions.mdx]: Modified to mention k8s not available from UI as this is now available for Servers

**Comment by stevenGravy** [comment]
> @stevenGravy it looks to me like this still has outstanding review items to be addressed by the author.

@alexfornuto Please take a look.  Comments addressed and added in web ui joining that is now available.

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 8]
```diff
@@ -65,27 +65,41 @@ The following fields are optional for `require_session_join`:
...
 
 The policy below specifies that the `prod-access` role must have a minimum of
 two users with the role `auditor` and the mode `moderator` present in the
 session to start it. The policy applies to SSH and Kubernetes sessions only.
 
-When a user with this require policy starts a session, it will be pending until
-the policy is fulfilled. 
+This policy requires that at least one user with the `auditor` role is present
+as a moderator for SSH or Kubernetes sessions to start. That is applied for
+Servers and Kubernetes resources labeled `env: prod`. The session will pend starting
```
Comment (zmb3) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:76]:
```suggestion
Servers and Kubernetes resources labeled `env: prod`. The session will not start
```

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 9]
```diff
@@ -105,7 +119,12 @@ The following are required options for `join_sessions`:
 |`kinds`|`[]`[Session kind](#session-kinds)|The kind of session that the policy applies to|
 |`modes`|`[]`[Participant mode](#participant-modes)|The participant mode that applies to the user joining the Moderated Session under this policy|
 
-#### Example
+<Admonition type="note">
+Joining from the UI is available for Server for SSH sessions.  Kubernetes are only joinable by CLI currently.
```
Comment (zmb3) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:123]:
```suggestion
Joining from the UI is available for SSH sessions.  Kubernetes sessions can only be joined from the CLI.
```

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 10]
```diff
@@ -124,6 +144,51 @@ spec:
...
+Teleport > Waiting for required participants...
+Teleport > User alice joined the session.
+Teleport > Connecting to prod.teleport.example.com over SSH
+
+ubuntu@prod.teleport.example.com % 
+```
+
+Here is an example of joining from the UI that is available for Server sessions.
+
+![Join Server Session from UI](../../../img/access-controls/joinsession.png)
```
Comment (zmb3) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:183]: We're about to change this, so I would either remove this section if you want to merge this now, or wait 1-2 weeks for an updated screenshot.

Comment (stevenGravy) [reply docs/pages/access-controls/guides/moderated-sessions.mdx:183]: Thanks let's wait then

**Comment by zmb3** [comment]
@rudream I assigned this to you so you can include an up-to-date screenshot after the join menu redesign.

**Comment by rudream** [comment]
@zmb3 @stevenGravy I created a PR that updates the screenshot. I kept the file name the same so it should work with the changes in this PR. https://github.com/gravitational/teleport/pull/25659

**Comment by stevenGravy** [comment]
> @zmb3 @stevenGravy I created a PR that updates the screenshot. I kept the file name the same so it should work with the changes in this PR. #25659
Thanks, good to approve then? @zmb3

**Comment by zmb3** [comment]
There's no need to add the image here if Yassine's PR will just replace it. Can we merge Yassine's branch here instead?

**Comment by rudream** [comment]
> There's no need to add the image here if Yassine's PR will just replace it. Can we merge Yassine's branch here instead?

@zmb3 @stevenGravy Instead of adding a new screenshot here, can we just reference the same one that is used in the [Web UI docs](https://goteleport.com/docs/ver/13.x/connect-your-client/web-ui/)? That is the one I am updating in my PR.

**Comment by stevenGravy** [comment]
> > > There's no need to add the image here if Yassine's PR will just replace it. Can we merge Yassine's branch here instead?
> > 
> > 
> > @zmb3 @stevenGravy Instead of adding a new screenshot here, can we just reference the same one that is used in the [Web UI docs](https://goteleport.com/docs/ver/13.x/connect-your-client/web-ui/)? That is the one I am updating in my PR.
> 
> Yes I can do that.

Done!

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 11]
```diff
@@ -64,27 +64,41 @@ The following fields are optional for `require_session_join`:
...
 
 The policy below specifies that the `prod-access` role must have a minimum of
 two users with the role `auditor` and the mode `moderator` present in the
 session to start it. The policy applies to SSH and Kubernetes sessions only.
 
-When a user with this require policy starts a session, it will be pending until
-the policy is fulfilled. 
+This policy requires that at least one user with the `auditor` role is present
+as a moderator for SSH or Kubernetes sessions to start. That is applied for
+Servers and Kubernetes resources labeled `env: prod`. The session will not start
```
Comment (ptgott) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:75]:
```suggestion
servers and Kubernetes resources labeled `env: prod`. The session will not start
```
"server" is not a proper noun

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 12]
```diff
@@ -64,27 +64,41 @@ The following fields are optional for `require_session_join`:
...
+until the policy is fulfilled.
 
 ```yaml
 kind: role
 metadata:
   name: prod-access
+version: v5
```
Comment (ptgott) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:82]: We should use v6 with the `kubernetes_resources` key

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 13]
```diff
@@ -114,6 +133,7 @@ as a moderator or observer.
 kind: role
 metadata:
   name: auditor
+version: v5
```
Comment (ptgott) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:136]: We should use v6

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 14]
```diff
@@ -123,6 +143,51 @@ spec:
...
+  - t: Forcefully terminate the session (moderators only)
+Teleport > User jeff joined the session.
+Teleport > Waiting for required participants...
+Teleport > User alice joined the session.
+Teleport > Connecting to prod.teleport.example.com over SSH
+
+ubuntu@prod.teleport.example.com % 
+```
+
+Here is an example of joining from the UI that is available for Server sessions.
```
Comment (ptgott) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:180]:
```suggestion
Here is an example of joining from the UI that is available for server sessions.
```

File: docs/pages/access-controls/guides/moderated-sessions.mdx [hunk 15]
```diff
@@ -123,6 +143,51 @@ spec:
...
+
+![Join Server Session from UI](../../../img/webui-active-session.png)
+
+<Admonition title="Note" type="tip">
+
+Joining as a moderator and observer is available for Servers in the UI Console.
```
Comment (ptgott) [inline docs/pages/access-controls/guides/moderated-sessions.mdx:186]: - "Servers" should be "servers", since it's not a proper noun
- What is the UI Console?

File: docs/pages/access-controls/guides/moderated-sessions.mdx (hunk 13 above)
Comment (stevenGravy) [reply docs/pages/access-controls/guides/moderated-sessions.mdx]: updated @ptgott

File: docs/pages/access-controls/guides/moderated-sessions.mdx (hunk 15 above)
Comment (stevenGravy) [reply docs/pages/access-controls/guides/moderated-sessions.mdx]: This meant Web UI but it's redundant since we're showing the joining of servers.  Removed

**Review by ptgott — APPROVED** [review]
